charlib/characterizer/cell.py

```
@@ -139,7 +139,35 @@ def match_direction(pin_name):
+            storage_vars = [var,]
+            for pair in self.diff_pairs.values():
+                if pin == pair.noninverting_port_name:
+                    storage_vars.append(feedback_paths[pair.complement(pin)])
+            if len(storage_vars) < 2:
+                # No inverting output. Append inv to the end of the first var name
+                storage_vars.append(f'{var}inv')
+            var_string = ', '.join(storage_vars)
+            if self.clock: # ff group
+                storage_group = liberty.Group('ff', var_string)
+                storage_group.add_attribute('next_state', self.functions[pin].expression)
+                storage_group.add_attribute('clocked_on', to_lib_expr(self.clock))
```

Ruff (0.15.17)

[error] Undefined name `to_lib_expr`

(F821)